File: frontend/src/api/parityBaseline.ts
import { apiClient, type createApiClient } from "./client";
import type { ParityBaselineRequest, ParityBaselineResponse } from "@/types/parityBaseline";
type Client = ReturnType<typeof createApiClient>;
const path = (runId: string) => `/api/v1/runs/${encodeURIComponent(runId)}/discovery/parity-baseline`;
export const captureParityBaseline = (runId: string, request: ParityBaselineRequest, client: Client = apiClient) => client.post<ParityBaselineResponse>(path(runId), request);
export const getParityBaseline = (runId: string, client: Client = apiClient) => client.get<ParityBaselineResponse>(path(runId));

File: backend/app/api/routes/parity_baseline.py
from fastapi import APIRouter, Depends, Request
from app.api.errors import error_response
from app.api.parity_baseline_contracts import ParityBaselineCaptureRequest, ParityBaselineResponse
from app.services.parity_baseline_evidence_application_service import (
    ParityBaselineEvidenceApplicationService,
    ParityBaselineEvidenceError,
)

router = APIRouter(prefix="/runs", tags=["parity-baseline"])


def service():
    return ParityBaselineEvidenceApplicationService()


@router.post("/{run_id}/discovery/parity-baseline", response_model=ParityBaselineResponse)
def capture(run_id: str, request: ParityBaselineCaptureRequest, http_request: Request, s=Depends(service)):
    try:
        return s.capture(run_id, request)
    except ParityBaselineEvidenceError as error:
        return error_response(http_request, status_code=error.status_code, error_code=error.code, message=error.message)


@router.get("/{run_id}/discovery/parity-baseline", response_model=ParityBaselineResponse)
def get(run_id: str, http_request: Request, s=Depends(service)):
    return s.get(run_id) or error_response(
        http_request,
        status_code=404,
        error_code="PARITY_BASELINE_NOT_FOUND",
        message="Parity baseline evidence was not found.",
    )
